src/pickup.cpp

```diff
@@ -169,6 +181,44 @@ interact_results interact_with_vehicle( vehicle *veh, const tripoint &pos,
             veh_tool( "hotplate" );
             return DONE;
 
+        case USE_WASHMACHINE: {
+            bool detergent_is_enough = g->u.crafting_inventory().has_charges( "detergent", 5 );
+            bool filthy_items = false;
+            for( auto &i : veh->get_items( veh->part_with_feature( veh_root_part, "WASHING_MACHINE" ) ) ) {
+                static const std::string filthy( "FILTHY" );
+                if( i.has_flag( filthy ) ) {
+                    i.bday = calendar::turn.get_turn();
+                    filthy_items = true;
+                }
+            }
+
+            for( auto e : veh->get_parts( "WASHING_MACHINE" ) ) {
+                if( e->enabled ) {
+                    e->enabled = false;
+                    add_msg( m_bad,
+                             _( "You turn the washing machine off before it's finished the program, and open its lid." ) );
+                } else if( veh->fuel_left( "water" ) < 24 ) {
+                    add_msg( m_bad, _( "You need 24 charges of water to fill the washing machine." ) );
+                } else if( !detergent_is_enough ) {
+                    add_msg( m_bad, _( "You need 5 charges of detergent for the washing machine." ) );
+                } else if( !filthy_items ) {
+                    add_msg( m_bad,
+                             _( "There are only non-filthy items in the washing machine.  There is no need to wash them." ) );
```

Comment:
@Coolthulhu says that player must wash only filthy items, you are telling right the opposite. I don't know what to prefer :-)